src/cmap/worker.rs

```diff
@@ -513,13 +513,11 @@ impl ConnectionPoolWorker {
                 handler.handle_pool_cleared_event(event);
             });
 
-            if !self.generation.is_load_balanced() {
```

Comment:
I swear that some part of the spec said that the queue shouldn't be drained in load-balancing mode, but I was apparently hallucinating that.

Reply:
Yep - https://github.com/mongodb/mongo-rust-driver/blob/master/src/test/spec/retryable_reads.rs#L74 failed (the second checkout succeeds rather than fails), and the load balancer spec [specifically calls out](https://github.com/mongodb/specifications/blob/master/source/load-balancers/tests/README.rst#id5) retryable reads tests.

Reply:
Ah okay, so that test is specifically verifying the pool paused behavior, which shouldn't apply to load balancers. The spec doesn't currently clarify this so I filed DRIVERS-1942 to update it. In the meantime, I think we can preserve the existing behavior of not clearing the WaitQueue on pool clear if in load balanced mode and just skip that individual test.

Reply:
Updated - needed to skip the equivalent test for retryable writes as well.